src/error.rs

```diff
@@ -139,6 +143,13 @@ pub enum ErrorKind {
     /// please report it to the warpgrapher team.
     FieldMissingResolverError(String, String),
 
+    /// Returned by the `From` implementation for a `String` object.
+    /// Conversion from `String`.
+    GenericError(String),
+
+    /// Returned by the engine for convenience when converting a json object containing an error message to a string. This error wraps the resulting string.
+    GraphQLError(String),
```

Comment:
I don't quite follow what this really means in concrete terms.  Maybe an example here, in the documentation?

Reply:
This is an error that comes from the pre_request_hook extensions, the error returned from that gets formatted into a GraphQL Json string and used to be returned directly from the Actix server. I created this error to wrap the resulting GraphQL Json string so we can return that from the engine. @shellforge do we have a specific error type coming from extensions that I can pass back directly? Or should I make this a specific error for PreRequestHookExtensionError?

Reply:
PreRequestHookExtensionError sounds appropriate.

Reply:
Replaced GraphQLError with extension specific ErrorKind